community/views.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Views for Community Resources
+"""
+
+# Imports #####################################################################
+
+from collections import defaultdict
+
+from django.core.urlresolvers import reverse
+from django.http import HttpResponse
+from django.template import loader
+
+from instance.models.openedx_appserver import OpenEdXAppServer
+from instance.models.mixins.secret_keys import OPENEDX_SECRET_KEYS, OPENEDX_SHARED_KEYS, generate_secret_key
+
+
+# Utils #######################################################################
+
+class ForcedDefaultDict(defaultdict):
+    """
+    A defaultdict that always returns True for __contains__
+    This is required to use a defaultdict as the root context dict
+    when rendering a django template.
+    """
+    def __contains__(self, key):
+        """ is key in this defaultdict? """
+        return True
+
+
+# Views #######################################################################
+
+def example_openstack_vars(_request):
+    """
+    Output some example ansible variables that contain no secrets
+
+    We cannot use the actual code from
+    OpenEdXAppServer.create_configuration_settings
+    because we can't control its context and it could easily contain
+    secrets from django settings, etc.
```

Comment:
Would it work, alternatively, to generate the file based on a vanilla `env` config? If there is no setting to leak, it wouldn't be a problem?